collect/collect.go

```diff
@@ -1,49 +1,58 @@
 package collect
 
 import (
-	"fmt"
-
+	v1beta1 "github.com/kubescape/storage/pkg/apis/softwarecomposition/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
 	"gopkg.in/yaml.v2"
 )
 
-type Severities struct {
-	Critical   Severity `yaml:"critical"`
-	High       Severity `yaml:"high"`
-	Low        Severity `yaml:"low"`
-	Medium     Severity `yaml:"medium"`
-	Negligible Severity `yaml:"negligible"`
-	Unknown    Severity `yaml:"unknown"`
-}
+var (
+	clusterCritical = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kubescape_controls_total_cluster_critical",
+		Help: "Total number of critical vulnerabilities in the cluster",
+	})
 
-type Severity struct {
-	All      int `yaml:"all"`
-	Relevant int `yaml:"relevant"`
-}
-type Spec struct {
-	Severities Severities `yaml:"severities"`
-}
-type VulnerabilitySummary struct {
-	Spec Spec `yaml:"spec"`
+	clusterHigh = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kubescape_controls_total_cluster_high",
+		Help: "Total number of high vulnerabilities in the cluster",
+	})
+
+	clusterMedium = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kubescape_controls_total_cluster_medium",
+		Help: "Total number of medium vulnerabilities in the cluster",
+	})
+
+	clusterLow = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kubescape_controls_total_cluster_low",
+		Help: "Total number of low vulnerabilities in the cluster",
+	})
+
+	clusterUnknown = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kubescape_controls_total_cluster_unknown",
+		Help: "Total number of unknown vulnerabilities in the cluster",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(clusterCritical)
+	prometheus.MustRegister(clusterHigh)
+	prometheus.MustRegister(clusterMedium)
+	prometheus.MustRegister(clusterLow)
+	prometheus.MustRegister(clusterUnknown)
 }
 
-func GetSeverityValues(yamlData []byte, severity string) (int, int, error) {
-	var summary VulnerabilitySummary
+func GetSeverityValues(yamlData []byte) (*v1beta1.WorkloadConfigurationScanSummary, error) {
+	var summary v1beta1.WorkloadConfigurationScanSummary 
 	if err := yaml.Unmarshal(yamlData, &summary); err != nil {
-		return 0, 0, err
+		return nil, err
 	}
+	return &summary, nil
+}
 
-	switch severity {
-	case "critical":
-		return summary.Spec.Severities.Critical.All, summary.Spec.Severities.Critical.Relevant, nil
-	case "high":
-		return summary.Spec.Severities.High.All, summary.Spec.Severities.High.Relevant, nil
-	case "low":
-		return summary.Spec.Severities.Low.All, summary.Spec.Severities.Low.Relevant, nil
-	case "medium":
-		return summary.Spec.Severities.Medium.All, summary.Spec.Severities.Medium.Relevant, nil
-	case "negligible":
-		return summary.Spec.Severities.Negligible.All, summary.Spec.Severities.Negligible.Relevant, nil
-	default:
-		return 0, 0, fmt.Errorf("unknown severity: %s", severity)
-	}
+func ProcessMetrics(summary *v1beta1.WorkloadConfigurationScanSummary) {
```

Comment:
I see you are using `WorkloadConfigurationScanSummary` for exposing metrics which are for the entire cluster. 
This object includes a summary for a single workload. If you want a summary for the entire cluster, you should use `ScopedConfigurationScanSummaryList`. The `ScopedConfigurationScanSummary` object exposes a summary for a single namespace. When the helm-chart will be ready, you would run `kubectl get configurationscansummaries` and you will get a list of `ScopedConfigurationScanSummary`, i.e a `ScopedConfigurationScanSummaryList`, where each object is a summary for a single namespace. So I think it would be better to accept a `ScopedConfigurationScanSummaryList` object as a parameter for this function, iterate over the objects and sum each severity. This will give you the total number for the entire cluster.

If you would like to expose metrics by namespaces, then you could accept an `ScopedConfigurationScanSummary` object to do that.

Reply:
Got it